### PR TITLE
MSFT.CSharp: Replace if-branch with expression.

### PR DESCRIPTION
**Pull request by JonHanna**

### CODE REVIEW

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs [hunk 1]
```diff
@@ -823,25 +819,15 @@ internal EXPR BindToField(EXPR pOptionalObject, FieldWithType fwt, BindingFlag b
...
-                        InAnonymousMethod())
-                    {
-                        isLValue = false;
-                    }
-                }
+                if (RespectReadonly() && fwt.Field().isReadOnly
+                    && (ContainingAgg() == null || !InMethod() || !InConstructor()
+                        || fwt.Field().getClass() != ContainingAgg() || InStaticMethod() != fwt.Field().isStatic
+                        || (pOptionalObject != null && !isThisPointer(pOptionalObject)) || InAnonymousMethod()))
+                    isLValue = false;
```
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs]: I don't know that this change is better... it's harder to read, at least for me.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs [hunk 2]
```diff
@@ -737,11 +737,7 @@ private void AddConversionsForArguments(ArgumentObject[] arguments)
                 // Make sure we're not setting ref for the receiver of a call - the argument
                 // will be marked as ref if we're calling off a struct, but we don't want 
                 // to persist that in our system.
-                bool isFirstParamOfCallOrInvoke = false;
-                if (i == 0 && IsBinderThatCanHaveRefReceiver(payload))
-                {
-                    isFirstParamOfCallOrInvoke = true;
-                }
+                bool isFirstParamOfCallOrInvoke = i == 0 && IsBinderThatCanHaveRefReceiver(payload);
```
Comment (karelz) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs]: I would really prefer parenthesis - at least around "i == 0" -> "(i == 0)", but I don't see it in any our guidelines :(, so feel free to ignore ...
@stephentoub do you know if it is intentionally left out of coding guidelines?

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs [hunk 3]
```diff
@@ -823,25 +819,15 @@ internal EXPR BindToField(EXPR pOptionalObject, FieldWithType fwt, BindingFlag b
 
             EXPRFIELD pResult;
             {
-                bool isLValue = false;
-                if ((pOptionalObject != null && pOptionalObject.type.IsPointerType()) || objectIsLvalue(pOptionalObject))
-                {
-                    isLValue = true;
-                }
+                bool isLValue = pOptionalObject != null && pOptionalObject.type.IsPointerType() || objectIsLvalue(pOptionalObject);
```
Comment (karelz) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs]: Please bring back the parenthesis around the && subexpression (left part of ||). It's easier to see && / || precedence IMO.
I would also love to see them around the != subexpr, but as above - that is not part of official coding style :(

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs (hunk 2 above)
Comment (stephentoub) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs]: > intentionally left out

Yes, in cases like this there's no need for the parens, with precedence being obvious.

**Comment by JonHanna** [comment]
Since my plan was to do a bunch of uncontroversial changes, I'm just going to close this.

**Pull request closed without merging**